app/core/views.py

```diff
@@ -36,14 +49,13 @@ def get_queryset(self):
         if is_admin:
             return Recruiter.objects.filter(company=company)
 
-        return Recruiter.objects.all()
+        return Recruiter.objects.none()
 
 
 class RecruiterDetailsManageView(RetrieveDestroyAPIView):
-    """ Retrieve or Delete User Profile """
+    """ Retrieve or Delete Recruiter Profile """
     serializer_class = RecruiterRetrieveListSerializer
-    authentication_classes = [authentication.TokenAuthentication]
-    permission_classes = [permissions.IsAuthenticated]
+    permission_classes = [permissions.IsAuthenticated, IsOwner]
 
     def get_object(self):
         return self.request.user.recruiter
```

Comment:
Should the `destroy` method for this class delete the Recruiter instance as well?

Reply:
get_object will only get called when retrieve (GET HTTP) request will be made for recruiter user

Reply:
In the `destroy` method below this `get_object` is overridden and doesn't delete the instance. I commented on `get_object` because the `destroy` method wasn't selectable.

Reply:
Yes, it should destroy both the user and recruiter instance. 
In `destroy()`, when user will be removed, because of `on_delete=cascade`, recruiter object will be deleted as well.
This view is intended to either retrieve single recruiter instance or delete single recruiter instance along with the related user instance.